lib/tracer/rooted.go

```diff
@@ -5,15 +5,23 @@ import (
 	"encoding/binary"
 	"hash/fnv"
 
+	"github.com/omni-network/omni/lib/netconf"
+
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/trace"
 )
 
-// StartChainHeight returns a context and span rooted to the chain+height.
+// StartChainHeight returns a context and span rooted to the network+network.Version+chain+height.
 // This creates a new trace root and should generally only by xprovider or cprovider.
-func StartChainHeight(ctx context.Context, chain string, height uint64, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
-	// Deterministic TraceID for chain+height. So all traces across all instances are correlated.
+func StartChainHeight(ctx context.Context, network netconf.ID, chain string, height uint64, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
+	// Deterministic TraceID for network+network.Version+chain+height.
+	// So all traces of the same block across all apps/instances of the same network are correlated.
+	// Note this only works for protected networks with consistent versions.
+	// Ephemeral network traces will not be correlated.
```

Comment:
Why do we want the predictable traceID? Can we correlate the traces by tags? I want to learn more about this use case.

Reply:
I basically want to view the spans of a "chain+block" across all our binaries in a single view. If grafana can do this using attributes, then we dont need this.

Reply:
* If you mean all spans to be part of the same trace graph - I don't think it is possible - at least I'm not aware.
* If you mean searching by all the spans related to "chain+block" (different trace graphs), then it is possible.

I don't know if we can extend CosmosSDN to generate a trace graph of a consenus round. 
What I mean is that we can start a root span in the ProposeBlock method and then its trace ID + span ID could be propagated throughout all other validators and their methods. Not sure if that makes sense.